Fix golangci-lint issues related to missing error checking

We should fix missing error checks because that will help us catch such checks when it matters. Also, we should remove `--disable errcheck` from `.github/golangci-lint.yml.`
```sh
% golangci-lint run ./...
scm/test_helper.go:16:23: Error return value of `sc.CreateOrganization` is not checked (errcheck)
	sc.CreateOrganization(context.Background(), &OrganizationOptions{
	                     ^
ci/docker.go:53:16: Error return value of `d.logger.Sync` is not checked (errcheck)
		d.logger.Sync()
		             ^
ci/docker.go:240:17: Error return value of `json.Unmarshal` is not checked (errcheck)
		json.Unmarshal([]byte(scanner.Text()), out)
		              ^
internal/qtest/test_helper.go:249:15: Error return value of `db.UpdateUser` is not checked (errcheck)
	db.UpdateUser(admin)
	             ^
ci/docker_test.go:174:10: Error return value of `os.Mkdir` is not checked (errcheck)
	os.Mkdir(filepath.Join(dir, "tests"), 0o700)
	        ^
ci/docker_test.go:175:10: Error return value of `os.Mkdir` is not checked (errcheck)
	os.Mkdir(filepath.Join(dir, "assignments"), 0o700)
	        ^
ci/docker_test.go:274:10: Error return value of `os.Mkdir` is not checked (errcheck)
	os.Mkdir(filepath.Join(dir, "tests"), 0o700)
	        ^
web/hooks/pull_request.go:62:26: Error return value of `wh.db.UpdatePullRequest` is not checked (errcheck)
		wh.db.UpdatePullRequest(pullRequest)
		                       ^
assignments/assignments_test.go:215:22: Error return value of `db.UpdateAssignments` is not checked (errcheck)
	db.UpdateAssignments([]*qf.Assignment{assignment, assignment2})
	                    ^
assignments/assignments_test.go:277:22: Error return value of `db.UpdateAssignments` is not checked (errcheck)
	db.UpdateAssignments([]*qf.Assignment{assignment, assignment2})
	                    ^
main.go:66:19: Error return value of `logger.Sync` is not checked (errcheck)
	defer logger.Sync()
	                 ^
```

--- a/.github/workflows/golangci-lint.yml
+++ b/.github/workflows/golangci-lint.yml
@@ -24,6 +24,6 @@ jobs:
           version: latest
           skip-pkg-cache: true
           skip-build-cache: true
-          args: --disable errcheck --timeout 5m
+          args: --timeout 5m
           # Optional: show only new issues if it's a pull request. The default value is `false`.
           only-new-issues: true

--- a/.vscode/quickfeed-words.txt
+++ b/.vscode/quickfeed-words.txt
@@ -102,6 +102,7 @@ mattn
 maxage
 meling
 mkdir
+multierr
 newkey
 ngrok
 nocheck

--- a/assignments/assignments_test.go
+++ b/assignments/assignments_test.go
@@ -212,7 +212,10 @@ func TestUpdateCriteria(t *testing.T) {
 	}
 
 	// Update assignments. GradingBenchmarks should not be updated
-	db.UpdateAssignments([]*qf.Assignment{assignment, assignment2})
+	err := db.UpdateAssignments([]*qf.Assignment{assignment, assignment2})
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	for _, wantReview := range []*qf.Review{review, review2} {
 		gotReview, err := db.GetReview(&qf.Review{ID: wantReview.ID})
@@ -274,7 +277,10 @@ func TestUpdateCriteria(t *testing.T) {
 	}
 
 	// Update assignments. GradingBenchmarks should be updated
-	db.UpdateAssignments([]*qf.Assignment{assignment, assignment2})
+	err = db.UpdateAssignments([]*qf.Assignment{assignment, assignment2})
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	// Benchmarks should have been updated to reflect the removal of a benchmark and a criterion
 	gotBenchmarks, err = db.GetBenchmarks(&qf.Assignment{ID: assignment.ID, CourseID: course.ID})

--- a/ci/docker.go
+++ b/ci/docker.go
@@ -18,6 +18,7 @@ import (
 	"github.com/docker/docker/api/types/mount"
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/quickfeed/quickfeed/internal/multierr"
 	"go.uber.org/zap"
 )
 
@@ -49,10 +50,12 @@ func NewDockerCI(logger *zap.SugaredLogger) (*Docker, error) {
 
 // Close ensures that the docker client is closed.
 func (d *Docker) Close() error {
+	var syncErr error
 	if d.logger != nil {
-		d.logger.Sync()
+		syncErr = d.logger.Sync()
 	}
-	return d.client.Close()
+	closeErr := d.client.Close()
+	return multierr.Join(syncErr, closeErr)
 }
 
 // Run implements the CI interface. This method blocks until the job has been
@@ -237,7 +240,9 @@ func printInfo(logger *zap.SugaredLogger, rd io.Reader) error {
 	scanner := bufio.NewScanner(rd)
 	for scanner.Scan() {
 		out := &dockerJSON{}
-		json.Unmarshal([]byte(scanner.Text()), out)
+		if err := json.Unmarshal([]byte(scanner.Text()), out); err != nil {
+			return err
+		}
 		if out.Error != "" {
 			return errors.New(out.Error)
 		}

--- a/ci/docker_test.go
+++ b/ci/docker_test.go
@@ -171,8 +171,12 @@ func TestDockerEnvVars(t *testing.T) {
 
 	// dir is the directory to map into /quickfeed in the docker container.
 	dir := t.TempDir()
-	os.Mkdir(filepath.Join(dir, "tests"), 0o700)
-	os.Mkdir(filepath.Join(dir, "assignments"), 0o700)
+	if err := os.Mkdir(filepath.Join(dir, "tests"), 0o700); err != nil {
+		t.Error(err)
+	}
+	if err := os.Mkdir(filepath.Join(dir, "assignments"), 0o700); err != nil {
+		t.Error(err)
+	}
 
 	out, err := docker.Run(context.Background(), &ci.Job{
 		Name:     t.Name() + "-" + qtest.RandomString(t),
@@ -271,8 +275,12 @@ WORKDIR /quickfeed
 
 	// dir is the directory to map into /quickfeed in the docker container.
 	dir := t.TempDir()
-	os.Mkdir(filepath.Join(dir, "tests"), 0o700)
-	os.Mkdir(filepath.Join(dir, "assignments"), 0o700)
+	if err := os.Mkdir(filepath.Join(dir, "tests"), 0o700); err != nil {
+		t.Error(err)
+	}
+	if err := os.Mkdir(filepath.Join(dir, "assignments"), 0o700); err != nil {
+		t.Error(err)
+	}
 
 	xTestGo, err := os.ReadFile("testdata/tests/x_test.go")
 	if err != nil {

--- a/internal/multierr/multierr.go
+++ b/internal/multierr/multierr.go
@@ -1,0 +1,33 @@
+package multierr
+
+import (
+	"fmt"
+	"strings"
+)
+
+type multiErr []error
+
+func (errs multiErr) Error() string {
+	var b strings.Builder
+	if len(errs) == 1 {
+		return errs[0].Error()
+	}
+	for _, e := range errs {
+		fmt.Fprintf(&b, "%s\n", e)
+	}
+	return b.String()
+}
+
+func Join(errs ...error) error {
+	err := make(multiErr, 0, len(errs))
+	for _, e := range errs {
+		if e == nil {
+			continue
+		}
+		err = append(err, e)
+	}
+	if len(err) == 0 {
+		return nil
+	}
+	return err
+}

--- a/internal/multierr/multierr_test.go
+++ b/internal/multierr/multierr_test.go
@@ -1,0 +1,30 @@
+package multierr_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/quickfeed/quickfeed/internal/multierr"
+)
+
+func TestMultiErr(t *testing.T) {
+	nilErr := multierr.Join(nil)
+	if nilErr != nil {
+		t.Errorf("Join(nil) = %v, want <nil>", nilErr)
+	}
+	nilErr = multierr.Join(nil, nil, nil)
+	if nilErr != nil {
+		t.Errorf("Join(nil) = %v, want <nil>", nilErr)
+	}
+
+	wantErr1 := errors.New("a")
+	wantErr2 := errors.New("b")
+	err1 := multierr.Join(wantErr1)
+	if err1 == nil {
+		t.Errorf("Join(a) = %v, want %v", err1, wantErr1)
+	}
+	err2 := multierr.Join(wantErr1, wantErr2)
+	if err2 == nil {
+		t.Errorf("Join(a,b) = %v, want %v, %v", err2, wantErr1, wantErr2)
+	}
+}

--- a/internal/qtest/test_helper.go
+++ b/internal/qtest/test_helper.go
@@ -246,7 +246,9 @@ func PopulateDatabaseWithInitialData(t *testing.T, db database.Database, sc scm.
 	}
 	course.OrganizationID = org.GetID()
 	admin := CreateAdminUser(t, db, course.GetProvider())
-	db.UpdateUser(admin)
+	if err = db.UpdateUser(admin); err != nil {
+		return err
+	}
 	CreateCourse(t, db, admin, course)
 
 	repos, err := sc.GetRepositories(ctx, org)

--- a/main.go
+++ b/main.go
@@ -63,7 +63,7 @@ func main() {
 	if err != nil {
 		log.Fatalf("Can't initialize logger: %v", err)
 	}
-	defer logger.Sync()
+	defer func() { _ = logger.Sync() }()
 
 	db, err := database.NewGormDB(*dbFile, logger)
 	if err != nil {

--- a/scm/test_helper.go
+++ b/scm/test_helper.go
@@ -13,7 +13,7 @@ func TestSCMManager() *Manager {
 		&app.Config{},
 	}
 	sc := NewFakeSCMClient()
-	sc.CreateOrganization(context.Background(), &OrganizationOptions{
+	_, _ = sc.CreateOrganization(context.Background(), &OrganizationOptions{
 		Name: "testorg",
 		Path: "testorg",
 	})

--- a/web/hooks/pull_request.go
+++ b/web/hooks/pull_request.go
@@ -59,7 +59,10 @@ func (wh GitHubWebHook) handlePullRequestReview(payload *github.PullRequestRevie
 	// review is from a course teacher, do we mark the pull request as approved for QuickFeed.
 	if reviewer.IsTeacher() {
 		pullRequest.SetApproved()
-		wh.db.UpdatePullRequest(pullRequest)
+		if err := wh.db.UpdatePullRequest(pullRequest); err != nil {
+			wh.logger.Errorf("Failed to update pull request in database: %v", err)
+			return
+		}
 		wh.logger.Debugf("Pull request successfully approved for repository: %s", payload.GetRepo().GetFullName())
 	}
 }

